code-challenges.test.js

```diff
@@ -10,43 +10,169 @@
+//     Test Suites: 1 failed, 1 total
+//     Tests:       1 failed, 1 total
+
+const secretCode = (string) => {
+    let four = /a/ig
+    let three = /e/ig
+    let one = /i/ig
+    let zero = /o/ig
+    return string.replaceAll(four, 4).replaceAll(three, 3).replaceAll(one, 1).replaceAll(zero, 0)
+}
+/*
+Test Suites: 1 passed, 1 total
+Tests:       1 passed, 1 total
+I looked up if there was a way to replace a character with a number. 
+My search led me to the built-in method .replaceAll()
+.replaceAll() returns a string with all matches of a pattern that can be replaced by a replacement. 
+The pattern being used is regex, or regular expressions, in order to find each instance of the argument and replaced with a number.
+I am sure there is a way to make this code efficient, but I am unable to figure out a way to solve that issue.
+*/
+
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+describe("wordSearch", () => {
+    const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+    const letterA = "a"
+    // Expected output: ["Mango", "Apricot", "Peach"]
+    const letterE = "e"
+    // Expected output: ["Cherry", "Blueberry", "Peach"]
+    it("returns an array of all the words containing that particular letter", () => {
+        expect(wordSearch(fruitArray, letterA)).toEqual(["Mango", "Apricot", "Peach"])
+        expect(wordSearch(fruitArray, letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+    })
+
+    // ReferenceError: wordSearch is not defined
+//     Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 passed, 2 total
 
-const letterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
 
 // b) Create the function that makes the test pass.
 
+/*
+Create a function named wordSearch.
+The function will take in two parameters named array and letter.
+The argument that will be passed into the parameter array will be an array of words.
+The argument that will be passed into the parameter array will be variable that will contain one character inside of a string.
+Using the .filter() method, you can use the variable as the call back function
+The function will return a new array that contains only the words that match the character
+*/
+const wordSearch = (array, letter) => {
+    return array.filter((value) => value.toLowerCase().includes(letter.toLowerCase()))
+}
+/*
+Test Suites: 1 passed, 1 total
+Tests:       2 passed, 2 total
+Reading the documentation on the .filter() method, there is an example of how to use the method to find a string to uses the filter criteria and push it into a new array.
```

Comment:
Thank you for writing this here.  That is certainly an approach, however, I am partial to the approach you have here.  Since filter already creates a sub-array, there is no need to create a new array and push to that the values.  If using a for loop, that would be the way to go though.